src/settings.ts

```diff
@@ -0,0 +1,91 @@
+import { app } from 'electron'
+import { writeFile, open } from 'fs/promises'
+import path from 'node:path'
+import { AppSettings } from './schemas/appSettings'
+import { version } from '../package.json'
+
+const defaultSettings: AppSettings = {
+  appVersion: version,
```

Comment:
This file contains helpers for handling the settings JSON, including the default configuration that will be set when no file is found.

Currently, we don't support any type of _migration_ that allows the settings file to be backward compatible in case we introduce breaking changes. I'm including `appVersion` here so we can keep track of the schema between releases.